### PR TITLE
mcux: add support of RT1010

### PR DESCRIPTION
**Pull request by jhqian**
ext: hal: nxp: mcux: add device files for RT1010

Add mcux 2.6.0 drivers and device header files for mimxrt1010.
Update
several drivers that were already imported for other SoCs but
also apply to mimxrt1010.

Origins: NXP MCUxpresso SDK 2.6.0
URL: mcuxpresso.nxp.com
Maintained-by: External

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

### CODE REVIEW

**Comment by jhqian** [comment]
> > Looks fine, but i can't trace the history of SDK_DelayAtLeastUs
> 
> I agree with @agansari . Why can't we pick up the `SDK_DelayAtLeastUs` change with an updated SDK version?
> 
> Please also update `mcux/README`

SDK_DelayAtLeastUs is there, but moved to fsl_common.c. I'll update the README

File: mcux/README [hunk 1]
```diff
@@ -16,7 +16,8 @@ Status:
    SoC		Version (Release Date)
    ----------------------------------------------------------------------------
    LPC54114	SDK 2.6.0 (2019-06-14)
-   LPC55S69	SDK 2.6.3 (2019-09-26)
+   LPC55S69	SDK 2.6.3 (2019-06-26)
```
Comment (agansari) [inline mcux/README]: bad merge, your skd version is older

Comment (jhqian) [reply mcux/README]: sorry about that, will fix.

**Comment by agansari** [comment]
@jhqian I don't get this commit "mcux: remove SDK_DelayAtLeastUs from clock driver".
Did you use `mcux/scripts/import_mcux_sdk.py` to add board's sdk? This should've removed SDK_DelayAtLeastUs.

**Comment by jhqian** [comment]
> @jhqian I don't get this commit "mcux: remove SDK_DelayAtLeastUs from clock driver".
> Did you use `mcux/scripts/import_mcux_sdk.py` to add board's sdk? This should've removed SDK_DelayAtLeastUs.

I used that script to import the SDK package. the reason the remove that function is that in the latest SDK code base that function has been moved to fsl_common.h from soc specific clock driver, fsl_clock.c.

**Comment by agansari** [comment]
> SDK code base that function has been moved to fsl_common.h from soc specific clock driver, fsl_clock.c.

I assume fsl_common should have been imported by the script as it's part of sdk's driver folder. Am I wrong?

**Comment by jhqian** [comment]
> > SDK code base that function has been moved to fsl_common.h from soc specific clock driver, fsl_clock.c.
> 
> I assume fsl_common should have been imported by the script as it's part of sdk's driver folder. Am I wrong?

the updated fsl_common was imported correctly. And it caused the problem, since there are now two implementations of SDK_DelayAtLeastUs in fsl_clock.c and fsl_common.c for all the other soc in zephyr.

**Comment by agansari** [comment]
> > > SDK code base that function has been moved to fsl_common.h from soc specific clock driver, fsl_clock.c.
> > 
> > 
> > I assume fsl_common should have been imported by the script as it's part of sdk's driver folder. Am I wrong?
> 
> the updated fsl_common was imported correctly. And it caused the problem, since there are now two implementations of SDK_DelayAtLeastUs in fsl_clock.c and fsl_common.c for all the other soc in zephyr.

Okay, so fsl_common was imported okay for 1010 or RT, but created problems in other SoC or families. You chose to do it by hand and not import many more sdks.
Thank you for your explanation.

**Comment by jhqian** [comment]
> > > > SDK code base that function has been moved to fsl_common.h from soc specific clock driver, fsl_clock.c.
> > > 
> > > 
> > > I assume fsl_common should have been imported by the script as it's part of sdk's driver folder. Am I wrong?
> > 
> > 
> > the updated fsl_common was imported correctly. And it caused the problem, since there are now two implementations of SDK_DelayAtLeastUs in fsl_clock.c and fsl_common.c for all the other soc in zephyr.
> 
> Okay, so fsl_common was imported okay for 1010 or RT, but created problems in other SoC or families. You chose to do it by hand and not import many more sdks.
> Thank you for your explanation.

right. That patch just "manually" solves the problem on the other SoCs.

File: mcux/drivers/imx/fsl_common.h [hunk 2]
```diff
@@ -38,8 +38,8 @@
 
 /*! @name Driver version */
 /*@{*/
-/*! @brief common driver version 2.0.1. */
-#define FSL_COMMON_DRIVER_VERSION (MAKE_VERSION(2, 1, 0))
+/*! @brief common driver version 2.2.0. */
+#define FSL_COMMON_DRIVER_VERSION (MAKE_VERSION(2, 2, 0))
```
Comment (MaureenHelm) [inline mcux/drivers/imx/fsl_common.h]: How are we getting a newer version of fsl_common.h and other drivers, when RT1010 has SDK 2.6.0 and RT1060 has SDK 2.6.2?

Comment (jhqian) [reply mcux/drivers/imx/fsl_common.h]: RT1060 has several post RFP releases to add wifi support

**Comment by MaureenHelm** [comment]
@jhqian will SDK rel 11 include the SDK_DelayAtLeastUs changes?

**Comment by jhqian** [comment]
> @jhqian will SDK rel 11 include the SDK_DelayAtLeastUs changes?

yes

File: mcux/drivers/imx/fsl_adc.h [hunk 3]
```diff
@@ -20,7 +20,7 @@
  * Definitions
  ******************************************************************************/
 /*! @brief ADC driver version */
-#define FSL_ADC_DRIVER_VERSION (MAKE_VERSION(2, 0, 3)) /*!< Version 2.0.3. */
+#define FSL_ADC_DRIVER_VERSION (MAKE_VERSION(2, 0, 2)) /*!< Version 2.0.2. */
```
Comment (MaureenHelm) [inline mcux/drivers/imx/fsl_adc.h]: Please check that we're not reverting to older drivers. I suspect that we probably only want to import the devices and boards folders, and skip everything in drivers.

Comment (jhqian) [reply mcux/drivers/imx/fsl_adc.h]: revised

**Pull request merged**